client/dashboard/dashboard-charts/index.js

```diff
@@ -109,14 +119,56 @@ class DashboardCharts extends Component {
 		);
 	}
 
+	renderIntervalSelector() {
+		const allowedIntervals = getAllowedIntervalsForQuery( this.props.query );
+		if ( ! allowedIntervals || allowedIntervals.length < 1 ) {
+			return null;
+		}
+
+		const intervalLabels = {
+			hour: __( 'By hour', 'wc-admin' ),
+			day: __( 'By day', 'wc-admin' ),
+			week: __( 'By week', 'wc-admin' ),
+			month: __( 'By month', 'wc-admin' ),
+			quarter: __( 'By quarter', 'wc-admin' ),
+			year: __( 'By year', 'wc-admin' ),
+		};
+
+		return (
+			<SelectControl
+				className="woocommerce-chart__interval-select"
+				value={ this.state.interval }
```

Comment:
Should we check if the interval in state is in the allowed intervals list? Right now this relies on `ReportChart`'s use of `getIntervalForQuery()` to pass the proper interval to the `Chart` component.

Reply:
I like the current method because the report chart does filter out intervals that aren't allowed, but we can keep a user's preferences even if the current list does not allow it.

For example, if they select 'Quarter' for the interval and switch to a shorter time period that doesn't allow quarterly charting, they still see the chart without issue and when they switch back to a period that allows quarters, their interval will be switched back to quarters.